jgrapht-core/src/test/java/org/jgrapht/perf/shortestpath/DijkstraShortestPathPerformanceTest.java

```diff
@@ -0,0 +1,286 @@
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.perf.shortestpath;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.interfaces.ShortestPathAlgorithm;
+import org.jgrapht.alg.shortestpath.ALTAdmissibleHeuristic;
+import org.jgrapht.alg.shortestpath.AStarShortestPath;
+import org.jgrapht.alg.shortestpath.BidirectionalDijkstraShortestPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.generate.GnpRandomGraphGenerator;
+import org.jgrapht.generate.GraphGenerator;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.DirectedWeightedPseudograph;
+import org.jgrapht.graph.IntegerVertexFactory;
+import org.jgrapht.traverse.ClosestFirstIterator;
+import org.jgrapht.util.StopWatch;
+import org.openjdk.jmh.runner.RunnerException;
+
+import junit.framework.TestCase;
+
+/**
+ * A small benchmark comparing Dijkstra like algorithms. The benchmark creates a random graph and
+ * computes all-pairs shortest paths.
+ * 
+ * @author Dimitrios Michail
+ */
+public class DijkstraShortestPathPerformanceTest
+    extends TestCase
+{
+    private static final int PERF_BENCHMARK_VERTICES_COUNT = 250;
+    private static final double PERF_BENCHMARK_EDGES_PROP = 0.3;
+    private static final int WARMUP_REPEAT = 5;
+    private static final int REPEAT = 10;
+    private static final long SEED = 13l;
+
+    private static abstract class BenchmarkBase
+    {
+        protected Random rng = new Random(SEED);
+        protected GraphGenerator<Integer, DefaultWeightedEdge, Integer> generator = null;
+        protected DirectedGraph<Integer, DefaultWeightedEdge> graph;
+
+        abstract ShortestPathAlgorithm<Integer, DefaultWeightedEdge> createSolver(
+            Graph<Integer, DefaultWeightedEdge> graph);
+
+        public void setup()
+        {
+            if (generator == null) {
+                // lazily construct generator
+                generator = new GnpRandomGraphGenerator<>(
+                    PERF_BENCHMARK_VERTICES_COUNT, PERF_BENCHMARK_EDGES_PROP, rng, false);
+            }
+
+            DirectedWeightedPseudograph<Integer, DefaultWeightedEdge> weightedGraph =
+                new DirectedWeightedPseudograph<>(DefaultWeightedEdge.class);
+            this.graph = weightedGraph;
+
+            generator.generateGraph(graph, new IntegerVertexFactory(), null);
+
+            for (DefaultWeightedEdge e : weightedGraph.edgeSet()) {
```

Comment:
In one of the tests, you run ALTBenchmark. Is the graph you constructed here feasible for this Admissible Heuristic? Shouldn't the graph's weights obey the triangle inequality?